File: 1. Building Abstraction with Procedures/1.1 The Elements of Programming/square/square.exs
# SICP 1.1.4
# Compound Procedures

defmodule SICP do

  def square(x), do: x * x
  def sum_of_squares(x, y), do: square(x) + square(y)

end

IO.puts(SICP.square(4))
IO.puts(SICP.sum_of_squares(3, 4))

File: 1. Building Abstraction with Procedures/1.1 The Elements of Programming/square/square.js
function square(x) {
	return x * x;
}

function sum_of_squares(x, y) {
	return square(x) + square(y);
}

function f(a) {
	return sum_of_squares(a + 1, a * 2);
}

console.log("Square");
console.log(square(6));
console.log(sum_of_squares(2,3));
console.log(f(5));